big-o/problem-set-big-o.md

```diff
@@ -5,3 +5,549 @@
+* title: Big O
+##### !question
+
+Imagine a password of length _n_ that can contain only digit values (numbers).
+
+What will be the time complexity for [a brute force solution](https://en.wikipedia.org/wiki/Brute-force_attack) to break the password?
+
+##### !end-question
+##### !options
+
+* O(1)
+* O(log n)
+* O(n)
+* O(nlog n)
+* O(n^2)
+* O(n^3)
+* O(n!)
+* O(10^n)
+
+##### !end-options
+##### !answer
+
+* O(10^n)
+
+##### !end-answer
+##### !explanation
+
+To break the password of size _n_ where each value could be one of ten, 0-9 digits, we will need to try out each possibility starting with all n values being _0_. Then change one of the values to remaining 9 possibilities. For each of these, change another value to each of the remaining possibilities and so on.
+
+<br />
+
+The first of the n values could be any of 0 through 9. For each of these, we try the 10 possible values for the second of the n values and so on. This would lead to _10 * 10 * ... n times_ or in other words _10 ^ n_ possibilities to explore. Hence the time complexity will be _O(10 ^ n)_ or _exponential_.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 12 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: GdZGf5
+* title: Big O
+##### !question
+
+A traveling salesperson wants to visit _n_ cities. They can start the journey at any city and must visit each city once.
+
+How many different possibilities exist of the order in which they could visit all the _n_ cities?
+
+##### !end-question
+##### !options
+
+* O(1)
+* O(log n)
+* O(n)
+* O(nlog n)
+* O(n^2)
+* O(n^3)
+* O(n!)
+* O(10^n)
+
+##### !end-options
+##### !answer
+
+* O(n!)
+
+##### !end-answer
+##### !explanation
+
+We start with choosing one of the _n_ cities to be the starting cities. There _n_ possible options for this. For each such choice of first cities, the are _n - 1_ options to select the second city to visit. Then, _n - 2_ options to select the third city to vist and so on until only one city remains to visit.
+
+<br />
+
+Let's consider an example: Let's say _n_ is 3 and the possible cities are Atlanta, Boston and Chicago. Here's how we can explore the problem. We pick one of the three cities to be the first city. Let's say Atlanta. Then, for the next city to visit, there are two options, Boston or Chicago. Let's say we pick Boston then only one city remains unvisited i.e. Chicago. To try out all possible such options would look like to following:
+
+1. Atlanta  -> Boston -> Chicago
+1. Atlanta -> Chicago -> Boston
+1. Boston -> Atlanta -> Chicago
+1. Boston -> Chicago -> Atlanta
+1. Chicago -> Atlanta -> Boston
+1. Chicago -> Boston -> Atlanta
+
+This is a total of 6 possibilities, which is the same as _3 * 2 * 1_ or _3!_ i.e. _3 factorial_.
+
+<br />
+
+Generalizing to _n_ cities, this would be _n * (n-1) * (n-2) * ... * 1_ or _n!_ i.e. _n factorial_. In Big O terms, the time complexity will be _O(n!)_ or _factorial_.
```

Comment:
```suggestion
Generalizing to _n_ cities, this would be _n * (n-1) * (n-2) * ... * 1_ or _n!_, i.e. _n factorial_. In Big O terms, the time complexity will be _O(n!)_ or _factorial_.
```